lib/rules/indent.js

```diff
@@ -375,12 +376,25 @@ module.exports = {
                 lastToken = sourceCode.getTokenBefore(lastToken);
             }
 
+            const textBeforeClosingParenthesis = sourceCode.getText(lastToken, lastToken.loc.start.column).slice(0, -1);
+            const indentChar = indentType === "space" ? " " : "\t";
+            const replaceRegExp = new RegExp(indentChar, "g");
+            const textLenWithoutIndentation = textBeforeClosingParenthesis.replace(replaceRegExp, "").length;
             const endIndent = getNodeIndent(lastToken, true);
 
-            if ((endIndent.goodChar !== lastLineIndent || endIndent.badChar !== 0) && isNodeFirstInLine(node)) {
+            /* we consider both of these cases legit:
+                return ((bar === 1 || bar === 2) &&
+                  (z === 3 || z === 4)
+                );
+                return ((bar === 1 || bar === 2) &&
+                  (z === 3 || z === 4));
+              */
+            if ((((endIndent.goodChar !== firstLineIndent && !textLenWithoutIndentation)
+                    || (textLenWithoutIndentation && endIndent.goodChar !== lastLineIndent))
```

Comment:
I think we could replace the check with:

```js
if (!sourceCode.getText().slice(lastToken.range[0] - lastToken.loc.start.column, lastToken.range[0]).trim()) {
  // report node
}
```

This will make sure that we only report the `)` token if it is first in the line (i.e. there are no other tokens before `)` that are on the same line).